tests/indicators/Test.Smma.cs

```diff
@@ -0,0 +1,47 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Skender.Stock.Indicators;
+
+namespace Internal.Tests
+{
+    [TestClass]
+    public class Smma : TestBase
+    {
+        [TestMethod]
+        public void Standard()
+        {
+            int lookbackPeriod = 20;
+
+            List<SmmaResult> results = Indicator.GetSmma(history, lookbackPeriod)
+                .ToList();
+
+            // assertions
+
+            // proper quantities
+            // should always be the same number of results as there is history
+            Assert.AreEqual(502, results.Count);
+            Assert.AreEqual(502 - lookbackPeriod + 1, results.Where(x => x.Smma != null).Count());
+
+            // sample value
```

Comment:
I usually like to test the last `null` to ensure it starts on time: `Assert.IsNull(results[18].Smma)` 

As a side note, I've typically been doing explicit casts `(decimal)` rather than using `.Value`.  Both work!  I've done it explicitly mostly due to my inability to determine which is better in this scenario.  I see you had done `.Value` in the indicator code too.  What are your thoughts on it?

Reply:
Also, we should do a Convergence test to see how much runway this typically needs.  I suspect, if it's like EMA, it will need a minimum history of `N+100` with recommended `N+250` or `10×N` runway.  This smoothing approach takes a while to converge.

Reply:
> Also, we should do a Convergence test to see how much runway this typically needs. I suspect, if it's like EMA, it will need a minimum history of `N+100` with recommended `N+250` or `10×N` runway. This smoothing approach takes a while to converge.

No clue how to do a convergence test. Definitely a good thing to do to figure out the proper minimum history. I'm just not sure how!

Reply:
> I usually like to test the last initialization period for `null` to ensure it starts on time: `Assert.IsNull(results[18].Smma)`

👍 

> As a side note, I've typically been doing explicit casts `(decimal)` rather than using `.Value`. Both work! I've done it explicitly mostly due to my inability to determine which is better in this scenario. I see you had done `.Value` in the indicator code too. What are your thoughts on it?

I've only been working with nullable base types recently... I would instinctively want to do a cast as well, generally. On personal projects though (which is the majority of my programming experience!), I tend to condense lots of code (occasionally stacking ternary statements, chaining linq queries) so `.Value` is more straightforward in messy code. My new favorite is the `.HasValue` flag, so I can condense all of my null checks!!